### PR TITLE
AvocadoConfigParser

### PR DESCRIPTION
**Pull request by richtja**
This commit introduces `AvocadoConfigParser` which uses `configparser.ConfigParser` to parse INI files with avocado configuration. During the parsing process, it preserves the config files path, which is used for creating error messages when the files are badly formatted. And it removes the `config` file from jobdata directory since it is not used anymore.

Before this PR:
```
...
  File "<unknown>", line 2
    foo moo
    ^^^
SyntaxError: invalid syntax

```
After this PR:
```
...
    raise SyntaxError(
SyntaxError: Syntax error in config file ../testing/config, please check the value ["CANCEL", "SKIP", "FAIL"]
foo moo
```
Reference: #5857 #5643

### CODE REVIEW

File: avocado/core/settings.py [hunk 1]
```diff
@@ -283,6 +283,60 @@ def set_value(self, value, convert=False):
                 self._value = dst_type(value)
 
 
+class AvocadoConfigParser:
+    """AvocadoConfigParser uses ConfigPraser to parse config from INI files.
```
Comment (clebergnu) [inline avocado/core/settings.py:287]: Typo: "Praser"

File: avocado/core/settings.py [hunk 2]
```diff
@@ -283,6 +283,60 @@ def set_value(self, value, convert=False):
...
+
+    During the parsing process, it preserves the config files path.
+    The configuration from files is stored in dict like structure
```
Comment (clebergnu) [inline avocado/core/settings.py:290]: "in *a* dict-like structure"

**Review by clebergnu — CHANGES_REQUESTED** [review]
Hi @richtja , thanks for this! Other than the typo, this LGTM.

**Comment by richtja** [comment]
Hi @clebergnu thank you for your review. I fixed those typos with a `force-push`. Please have a look.

**Review by clebergnu — APPROVED** [review]
LGTM, thanks!

**Pull request merged**